README.md

```diff
@@ -342,6 +350,57 @@ Also specify the principal corresponding to the client account identified by `oa
 
 This is not a full set of available `oauth.*` properties. All the `oauth.*` properties described in the next chapter about [configuring the Kafka clients](#configuring-the-kafka-client) also apply to configuring the client side of inter-broker communication. 
 
+#### Enabling the re-authentication
+
+Access tokens expire after some time. The token validation for the purpose of session authentication is only performed immediately after the new connection from the client is established. 
+If using SimpleACLAuthorizer or no authorizer at all, then there is no further need for the access token after the validation, and by default the expiry of the token will not result in session closure or denial of access within the existing session.
+
+Since Kafka version 2.2 the Kafka brokers support a [re-authentication mechanism](https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate) allowing clients to update the token mid-session, without having to drop and re-establish the connection. 
+When a client sends the new access token, validation is performed on the broker as if a new connection was established.
+
+The mechanism has to be explicitly enabled on the Kafka broker by using the `connections.max.reauth.ms` property in `server.properties`:
+
+    connections.max.reauth.ms=3600000
+
+In this example the maximum time until next re-authentication is set to one hour.
+If the access token expires sooner than that, the re-authentication will be triggered sooner.
+
+The option can be specified per-listener. For example if you have a listener called `CLIENTS`, you can specify:
```

Comment:
Is it disabled by default? Or does it have some default value?

Reply:
It is actually disabled by default, which I didn't expect, but the reason is that it relies on additions to Kafka wire protocol introduced in Kafka 2.2, and the clients that don't understand it could break.

Reply:
Maybe you can add a note about it here? It wasn't completely clear to me.

Reply:
I added a little more text.